### PR TITLE
dwarf-fortress: freetype caveat

### LINKED ISSUE #418: Add building `freetype` with `--universal` to `dwarf-fortress` caveats
See Homebrew/homebrew#44570

### PR DESCRIPTION
**Pull request by timvisher**
Closes #418

/cc @bfontaine

### CODE REVIEW

**Comment by timvisher** [comment]
@bfontaine Let me know why the build is failing and I can try to fix it.

**Comment by bfontaine** [comment]
@timvisher Do you mind replacing the `sha1` with a `sha256`? It fails because of that :/

**Comment by timvisher** [comment]
@bfontaine Attempted. We'll see how that goes.

**Comment by dunn** [comment]
Should there be something like 
```ruby
depends_on "freetype" => [:optional, :universal]
```

**Comment by timvisher** [comment]
@dunn Sure? Let me know and I'll add it.

**Comment by dunn** [comment]
What does dwarf-fortress use freetype for?

**Comment by timvisher** [comment]
@dunn I'm assuming for rendering? I actually have no idea.

**Comment by dunn** [comment]
We should figure out if it's a mandatory or optional dependency before doing anything, then.

**Comment by timvisher** [comment]
@dunn It's mandatory afaict. Homebrew/homebrew#44570

**Comment by dunn** [comment]
Oh, cool. Then go ahead and add `depends_on "freetype" => :universal` and remove the caveat. Thanks!

File: dwarf-fortress.rb [hunk 1]
```diff
@@ -18,6 +18,10 @@ def install
   def caveats; <<-EOS.undent
     If you're using a retina display, the PRINT_MODE should
     be changed to STANDARD in #{libexec}/data/init/init.txt
+
+    Dwarf Fortress currently references freetype at a deprecated
```
Comment (mistydemeo) [inline dwarf-fortress.rb]: If it's compatible with a Homebrew-installed freetype, how about we make the dep mandatory and rewrite the install names in the binary to point it at that?

**Comment by mistydemeo** [comment]
Alternately, if the issue is that it needs X11 and the X11 path changed, maybe we can rewrite them to point to the actually-installed X11's prefix instead.

File: dwarf-fortress.rb (hunk 1 above)
Comment (timvisher) [inline dwarf-fortress.rb]: > If it's compatible with a Homebrew-installed freetype, how about we make the dep mandatory and rewrite the install names in the binary to point it at that?

I have no idea how to rewrite the binary to point to that. :P

**Comment by timvisher** [comment]
@mistydemeo @dunn The recipe did have a dep declared on X11, so I would think it doesn 'need X11'. Pretty much all I know about this is that I had a broken dwarf-fortress game which started working as soon as I reinstalled `freetype` with the `--universal` flag. I'm happy to either continue to have recommendations thrown my way or to have someone who knows more than I do take this over.

**Comment by timvisher** [comment]
ping :)

**Comment by bfontaine** [comment]
Sorry I modified the formula; could you please rebase the PR on top of `master`?

**Comment by timvisher** [comment]
@bfontaine Done.

**Comment by mistydemeo** [comment]
After a little experimentation, I have an alternate suggestion in #444; this uses `install_name_tool` to point SDL_ttf at the right library regardless of X11 prefix.

**Pull request closed without merging**